data-prepper-plugins/csv-processor/src/main/java/com/amazon/dataprepper/plugins/processor/csv/CSVProcessorConfig.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.processor.csv;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.constraints.NotNull;
+
+public class CSVProcessorConfig {
+    static final String DEFAULT_SOURCE = "message";
+    static final char DEFAULT_DELIMITER = ',';
+    static final char DEFAULT_QUOTE_CHARACTER = '\"'; // double quote
+    static final boolean DEFAULT_DELETE_HEADERS = true;
+
+    @JsonProperty("source")
+    @NotNull
+    private String source = DEFAULT_SOURCE;
+
+    @JsonProperty("delimiter")
+    @NotNull
+    private char delimiter = DEFAULT_DELIMITER;
+
+    @JsonProperty("delete_header")
+    @NotNull
+    private boolean deleteHeader = DEFAULT_DELETE_HEADERS;
+
+    @JsonProperty("quote_character")
+    @NotNull
+    private char quoteCharacter = DEFAULT_QUOTE_CHARACTER;
+
+    @JsonProperty("column_names_source_key")
+    private String columnNamesSourceKey;
+
+    @JsonProperty("column_names")
+    private String[] columnNames;
```

Comment:
Please use `List<String>`. Jackson will know how to deserialize a JSON array into a Java list.